systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaBridgeResource.java

```diff
@@ -5,124 +5,51 @@
     }
-
-    public static KafkaBridgeBuilder kafkaBridgeWithMetrics(String name, String clusterName, String bootstrap, int kafkaBridgeReplicas) {
-        KafkaBridge kafkaBridge = getKafkaBridgeFromYaml(PATH_TO_KAFKA_BRIDGE_METRICS_CONFIG);
-
-        return defaultKafkaBridge(kafkaBridge, name, clusterName, bootstrap, kafkaBridgeReplicas);
+    @Override
+    public KafkaBridge get(String namespace, String name) {
+        return kafkaBridgeClient().inNamespace(namespace).withName(name).get();
     }
-
-    private static KafkaBridgeBuilder defaultKafkaBridge(KafkaBridge kafkaBridge, String name, String kafkaClusterName, String bootstrap, int kafkaBridgeReplicas) {
-        return new KafkaBridgeBuilder(kafkaBridge)
-            .withNewMetadata()
-                .withName(name)
-                .withNamespace(ResourceManager.kubeClient().getNamespace())
-                .withClusterName(kafkaClusterName)
-            .endMetadata()
-            .editSpec()
-                .withBootstrapServers(bootstrap)
-                .withReplicas(kafkaBridgeReplicas)
-                .withNewInlineLogging()
-                    .addToLoggers("bridge.root.logger", Environment.STRIMZI_COMPONENTS_LOG_LEVEL)
-                .endInlineLogging()
-            .endSpec();
+    @Override
+    public void create(KafkaBridge resource) {
+        kafkaBridgeClient().inNamespace(resource.getMetadata().getNamespace()).createOrReplace(resource);
     }
-
-    public static KafkaBridge createAndWaitForReadiness(KafkaBridge kafkaBridge) {
-        KubernetesResource.deployNetworkPolicyForResource(kafkaBridge, KafkaBridgeResources.deploymentName(kafkaBridge.getMetadata().getName()));
-
-        TestUtils.waitFor("KafkaBridge creation", Constants.POLL_INTERVAL_FOR_RESOURCE_CREATION, CR_CREATION_TIMEOUT,
-            () -> {
-                try {
-                    kafkaBridgeClient().inNamespace(ResourceManager.kubeClient().getNamespace()).createOrReplace(kafkaBridge);
-                    return true;
-                } catch (KubernetesClientException e) {
-                    if (e.getMessage().contains("object is being deleted")) {
-                        return false;
-                    } else {
-                        throw e;
-                    }
-                }
-            }
-        );
-        return waitFor(deleteLater(kafkaBridge));
+    @Override
+    public void delete(KafkaBridge resource) throws Exception {
+        kafkaBridgeClient().inNamespace(resource.getMetadata().getNamespace()).withName(
+            resource.getMetadata().getName()).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
     }
-
-    public static KafkaBridge kafkaBridgeWithoutWait(KafkaBridge kafkaBridge) {
-        kafkaBridgeClient().inNamespace(ResourceManager.kubeClient().getNamespace()).createOrReplace(kafkaBridge);
-        return kafkaBridge;
+    @Override
+    public boolean isReady(KafkaBridge resource) {
```

Comment:
Does this return immediately? The call to `waitForResourceStatus` suggests it is not. And that means it should be named accordingly I think!

Reply:
I have changed it to -> `waitForReadiness`